src/v/config/configuration.cc

```diff
@@ -1689,7 +1689,18 @@ configuration::configuration()
       "cloud_storage_cache_size",
       "Max size of archival cache",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
-      20_GiB)
+      0,
+      property<uint64_t>::noop_validator,
+      legacy_default<uint64_t>(20_GiB, legacy_version{9}))
+  , cloud_storage_cache_size_percent(
```

Comment:
IIRC correctly, users can use a separate disk for the cloud storage cache. With that type of setup, this type of size allocation doesn't make sense. I don't think that's the case in our cloud, but I'd be curious if there's any self hosted customers that do this.

Reply:
Yes, that's correct. The percentage applies to either the shared data disk or the cache disk, but the default percent here is chosen assuming the shared disk setup.

We could do something automatic like switch it to 100% at runtime if a dedicated disk were chosen, but this seems like it might be confusing compared to solving this in documentation stating that this is the default in all cases, and if you use a dedicated disk, then you should consider increasing the value of this percentage tunable.